src/ccl_background.c

```diff
@@ -621,6 +621,122 @@ void ccl_cosmology_compute_distances(ccl_cosmology * cosmo, int *status)
   }
 }
 
+/* ----- ROUTINE: reverseArray ------
+INPUT: array, start, end
+TASK: Function to reverse arr[] from start to end.
+*/
+static void reverseArray(double arr[], int start, int end)
+{
+    double temp;
+    while (start < end)
+    {
+        temp = arr[start];
+        arr[start] = arr[end];
+        arr[end] = temp;
+        start++;
+        end--;
+    }
+}
+
+/* ----- ROUTINE: ccl_cosmology_distances_from_input ------
+INPUT: cosmology, scale factor array, comoving distance chi computed at the scale factor array values
+TASK: if not already there, make a table of comoving distances from an input array
+*/
+void ccl_cosmology_distances_from_input(ccl_cosmology * cosmo, int na, double a[], double chi_a[], double E_a[],
+   int *status)
+{
+  //Do nothing if everything is computed already
+  if(cosmo->computed_distances)
+    return;
+
+  // Allocate E(a), chi(a) and a(chi) splines
+  gsl_spline * E = gsl_spline_alloc(cosmo->spline_params.A_SPLINE_TYPE, na);
+  gsl_spline * chi = gsl_spline_alloc(cosmo->spline_params.A_SPLINE_TYPE, na);
+  gsl_spline * achi = gsl_spline_alloc(cosmo->spline_params.A_SPLINE_TYPE, na);
+
+  //Check for too little memory
+  if (a == NULL || E_a == NULL || chi_a == NULL || E == NULL || chi == NULL || achi == NULL) {
+    *status=CCL_ERROR_MEMORY;
+    ccl_cosmology_set_status_message(
+      cosmo, "ccl_background.c: ccl_cosmology_distances_from_input(): ran out of memory\n");
+  }
+
+  // FIXME: This is raised and I don't know why.
+  /*
+  //Check for messed up scale factor conditions
```

Comment:
Within `ccl_background.c` these two are used in the functions to compute distances and growth, to initialize the scale factor array at which points the various distances and functions will be computed. Since now these are going to be from input, perhaps it is only necessary to check for a few basic things (e.g. the scale factor is not larger than 1 or smaller than some pre-determined value)?

Reply:
OK, I think this is fine, since you're checking for this at the python level.

Reply:
Actually, related to this: ` cosmo->spline_params.A_SPLINE_MIN` is annoyingly used [here](https://github.com/LSSTDESC/CCL/blob/1f08a44733cafee5492d3d1914ecaf4ca7f09640/src/ccl_tracers.c#L458)

Could you update this variable within this function call to reflect the actual lower bound of the scale factor in the distance arrays?